src/components/scenes/TradeOptionSelectScene.tsx

```diff
@@ -0,0 +1,414 @@
+              )
+            })}
+          </>
+        )}
+      </SceneContainer>
+    </SceneWrapper>
+  )
+}
+
+const QuoteResult: React.FC<{
+  quotes: RampQuoteResult[]
+  onPress: (quote: RampQuoteResult) => Promise<void>
+  bestQuoteOverall?: RampQuoteResult
+}> = ({ quotes, onPress, bestQuoteOverall }) => {
+  const theme = useTheme()
+
+  // State for selected quote
+  const [selectedQuoteIndex, setSelectedQuoteIndex] = React.useState(0)
+  const selectedQuote = quotes[selectedQuoteIndex]
+
+  if (quotes.length === 0 || selectedQuote == null) {
+    return null
+  }
+
+  // Check if the currently selected quote is the best rate
+  const isBestRate =
+    bestQuoteOverall != null &&
+    selectedQuote.pluginId === bestQuoteOverall.pluginId &&
+    selectedQuote.paymentType === bestQuoteOverall.paymentType &&
+    selectedQuote.fiatAmount === bestQuoteOverall.fiatAmount
+
+  const fiatCurrencyCode = selectedQuote.fiatCurrencyCode.replace('iso:', '')
+
+  // Get the icon for the payment type
+  const paymentTypeIcon = getPaymentTypeIcon(selectedQuote.paymentType, theme)
+  const icon = paymentTypeIcon ?? { uri: selectedQuote.partnerIcon }
+
+  // Determine custom title rendering
+  const customTitleKey = paymentTypeToCustomTitleKey[selectedQuote.paymentType]
+  const defaultTitle = getPaymentTypeDisplayName(selectedQuote.paymentType)
+
+  // Render custom title based on payment type
+  let titleComponent: React.ReactNode
+  switch (customTitleKey) {
+    case 'applepay':
+      // Per Apple branding guidelines, "Pay with" is NOT to be translated
+      titleComponent = (
+        <TitleAppleContainer>
+          <TitleText numberOfLines={1}>
+            {/* eslint-disable-next-line react-native/no-raw-text */}
+            {'Pay with '}
+          </TitleText>
+          <TitleAppleLogo source={paymentTypeLogoApplePay} />
+        </TitleAppleContainer>
+      )
+      break
+    default:
+      titleComponent = <TitleText numberOfLines={1}>{defaultTitle}</TitleText>
+  }
+
+  // Handle provider press - show modal to select between providers
+  const handleProviderPress = async () => {
+    if (quotes.length <= 1) {
+      // No other providers to choose from
+      return
+    }
+
+    // Create items array for the RadioListModal
+    const items = quotes.map(quote => {
+      // Format the crypto amount for each provider
+      // const localeAmount = formatNumber(toFixed(quote.cryptoAmount, 0, 6))
+      const amount =
+        quote.direction === 'buy' ? quote.fiatAmount : quote.cryptoAmount
+      const currencyCode =
+        quote.direction === 'buy'
+          ? quote.fiatCurrencyCode.replace('iso:', '')
+          : quote.displayCurrencyCode
+      const text = `(${amount} ${currencyCode})`
+
+      return {
+        name: quote.pluginDisplayName,
+        icon: quote.partnerIcon, // Already full path
+        text
+      }
+    })
+
+    const selectedName = await Airship.show<string | undefined>(bridge => (
```

Comment:
Use a stable key for radio list selection.
`selected={selectedQuote.pluginId}`
`const selectedIndex = quotes.findIndex(q => q.pluginId === selectedName)`